dast/diff.py

```diff
@@ -0,0 +1,58 @@
+"""
+The main logic for the diff as it differs from DeepDiff
+"""
+from deepdiff import DeepDiff, DeepHash
+from deepdiff.diff import ListItemRemovedOrAdded, DiffLevel
+
+from dast.lcs import get_lcs
+
+class Diff(DeepDiff):
+    def _get_matching_pairs(self, level):
+        """
+        Given a level get matching pairs. This returns list of two tuples in the form:
+        [
+          (t1 index, t2 index), (t1 item, t2 item)
+        ]
+
+        Will compute the LCS between the two levels by using the DistanceMixin and a maximum cutoff threshold
+        """
+        t1_hashes = DeepHash(level.t1)
+        t1 = [t1_hashes[level.t1[i]] for i in range(len(level.t1))]
```

Comment:
this is not just `t1_hashes.values()`? since python 3.7 we can rely on the insert ordering

Reply:
The Deephash function contains all the hashes recursively down to the simplest objects, so it has some extra stuff. Plus I don't know which order they do it in (could be reverse for certain recursive function I suppose) so safer this way

Reply:
```suggestion
        t1 = [t1_hashes[obj]] for obj in level.t1]
```
I guess we can simplify it in this way